libs/dex/src/main/java/io/crate/data/CollectingBatchIterator.java

```diff
@@ -164,9 +164,11 @@ public boolean allLoaded() {
     }
 
     @Override
-    public void kill(@Nonnull Throwable throwable) {
-        onKill.accept(throwable);
-        killed = throwable;
+    public void kill(@Nonnull Throwable throwable, Type killType) {
+        onKill.accept(throwable, killType);
+        if (killType == Type.HARD) {
+            killed = throwable;
+        }
```

Comment:
I think this would deserve a comment on why `killed` is only set in the `Type.HARD` case

Reply:
Thanks, added some explanation